### PR TITLE
Reference a single diskIsMapped var, rather than copying it.

### PR DESCRIPTION
**Pull request by tylerwilliams**
The following conditions need to be true to hit this send on closed channel case:

- file must exist
- file must not be found in LRU
- WithPrefix must have been called before scanning finished 
- Channel must be closed -- meaning stuff is done being scanned but is being inserted into the LRU

In practice this means the window where this can happen is during the time between finishing scanning everything on disk, and before everything has been fully inserted into the LRU. Which explains why this is uncommon.

### CODE REVIEW

File: server/backends/disk_cache/disk_cache.go [hunk 1]
```diff
@@ -120,7 +120,8 @@ func (c *DiskCache) initializeCache() error {
 		return err
 	}
 	c.mu.Lock()
```
Comment (vadimberezniker) [inline server/backends/disk_cache/disk_cache.go:122]: Is this block necessary? It looks like c.diskIsMapped is already guaranteed to be false here.

File: server/backends/disk_cache/disk_cache.go [hunk 2]
```diff
@@ -98,8 +100,6 @@ func NewDiskCache(rootDir string, maxSizeBytes int64) (*DiskCache, error) {
 }
 
 func (c *DiskCache) WithPrefix(prefix string) interfaces.Cache {
-	c.mu.Lock()
```
[the remark this answers is not in the document]
Comment (tylerwilliams) [reply server/backends/disk_cache/disk_cache.go:101]: yeah gp

File: server/backends/disk_cache/disk_cache.go (hunk 1 above)
Comment (tylerwilliams) [reply server/backends/disk_cache/disk_cache.go]: nah, removed

File: server/backends/disk_cache/disk_cache.go [hunk 3]
```diff
@@ -164,7 +165,8 @@ func (c *DiskCache) initializeCache() error {
 		for _, record := range inFlightRecords {
 			c.l.Add(record.key, record)
 		}
-		c.diskIsMapped = true
+		mappedBool := true
```
[the remark this answers is not in the document]
Comment (tylerwilliams) [reply server/backends/disk_cache/disk_cache.go:167]: the next line updates the pointer -- or am I misunderstanding?

[the remark this answers is not in the document]
Comment (vadimberezniker) [reply server/backends/disk_cache/disk_cache.go:167]: Anyone that has called WithPrefix() before this point will have the pointer to the "false" boolean variable setup in NewCache. Updating c.diskIsMapped with a new pointer won't be visible to other instances of the cache.
I think you want `c.diskIsMapped = true` here.

[the remark this answers is not in the document]
Comment (tylerwilliams) [reply server/backends/disk_cache/disk_cache.go]: ahhhhh, I get what you're saying. Yeah. (*c.diskIsMapped = true)

**Review by vadimberezniker — APPROVED** [review]
LGTM

**Pull request merged**